contrib/format-hdf5/src/test/java/org/apache/drill/exec/store/hdf5/TestHDF5Format.java

```diff
@@ -98,9 +98,9 @@ public void testStarQuery() throws Exception {
 
     testBuilder()
       .sqlQuery("SELECT * FROM dfs.`hdf5/dset.h5`")
-      .unOrdered()
-      .baselineColumns("path", "data_type", "file_name", "int_data")
-      .baselineValues("/dset", "DATASET", "dset.h5", finalList)
+      .ordered()
```

Comment:
This might be the place to ask the question about schema. We have two distinct views of a data set. The general rule of the wildcard (`*`) is to return all available columns. Here, we special-case wildcard to mean "return metadata." This is, unfortunately, very non standard.

We need some way to express two views of the file. The same problem occurs for any database. We could even use if for JSON, CSV and other file formats.

The challenge is, how do we tell the query we want metadata and not data? In a normal DB, we query system tables. Perhaps we could jimmy up something in Drill:

```
SELECT * FROM sys.schema.dfs.`hdf5/dset.h5`
```

Or, maybe think of the table as a namespace, and have an optional `.schema` tail:

```
SELECT * FROM dfs.`hdf5/dset.h5`.schema
```

The point is not for you to implement this, or even to design the solution. Rather, the point is that the current solution is a hack, and that we need a better solution.

Reply:
Hey @paul-rogers 
I'd agree that this is a bit of a hack, but I think you misunderstood how this plugin works.  There is a config variable called `defaultPath` which when `null` returns metadata.  If this variable is set to an HDF5 path, you get the data (and no metadata). 

The issue is the rather unique nature of HDF5, as a file system within a file.  I do like the idea of treating the file as directory however, the metadata is really useful to the user in that some file paths are shortcuts, some are groups etc.   Also, datasets have attributes which can be useful. 

IMHO, this really blends the lines between storage and format plugin, so it's quite challenging to design this.  I am curious as to whether this actually gets used in the scientific community.

Reply:
@cgivre, you are right about my lack of understanding. This thing is a beast and I'm afraid I don't understand much of it. The config variable is also a hack: it means the user needs a separate config for each file and data set. The ease of use of that is perhaps not what it should be. Having a session option would be a bit better, but still not great. (Of course, Drill provides no way for a plugin to define a session option. Another good feature request.)

I posted the question on the dev list and learned that we probably can modify Drill to allow you to specify the data set name as part of the table path. And, we can work out how to extend the `DESCRIBE` statement to query for schema rather than data.

All of this is beyond the scope of this PR. If we use HDF5 as a prototype, we can see that we'd want to do similar things for other format or plugins.

Reply:
@paul-rogers 
Thanks as always for your responses.   I don't know that I'd use this plugin as an example of much, just because I don't think there are many other data formats like it. (Maybe I'm wrong about that).

I do think we should look at the LTSV and other EVF conversions and make incremental improvements to an "Easy EVF" interface that will reduce cut/paste code in format plugins and make them easier to write.

Reply:
In retrospect, the best way would probably be to do some sort of hybrid plugin that has access to the Calcite schema. That way you could add the HDF5 data path after the file.  If there isn't a data path.. you get metadata only.  However, I suspect that would be very difficult with the current architecture and would involve a lot of cut/paste and or extending from the file system plugin.

Reply:
Agreed. The point is not so much to change what you have here. Rather, it is to learn what more Drill should provide. Some other random examples of nested schemas are 1) REST calls, 2) Zip archives, 3) Iceberg files, 4) the JDBC storage plugin.

I'm sure these are all done either not at all (Zip, Iceberg) or via ad-hoc code (JDBC). The lesson is that Drill probably needs a way to support nested schemas; not all data sources are Parquet or JSON files on HDFS.

Reply:
I'm guessing that this isn't for this PR either?